tests/gcloud-testing-core/src/main/java/com/google/cloud/runtime/jetty/testing/AppDeployment.java

```diff
@@ -0,0 +1,68 @@
+package com.google.cloud.runtime.jetty.testing;
+
+import org.eclipse.jetty.toolchain.test.MavenTestingUtils;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Accessible AppDeployment information constants.
+ * <p>
+ * Will pull information from the system environment and also the
+ * active project's app.yaml in order to build the server deployed
+ * URI and variables.
+ * </p>
+ */
+public final class AppDeployment {
+  public static final String PROJECT_ID;
+  public static final String VERSION_ID;
+  public static final String SERVICE_ID;
+  public static final URI SERVER_URI;
+
+  static {
+    String projectId = System.getProperty("app.deploy.project");
+    String version = System.getProperty("app.deploy.version");
+
+    Objects.requireNonNull(projectId, "app.deploy.project");
+    Objects.requireNonNull(version, "app.deploy.version");
+
+    PROJECT_ID = projectId;
+    VERSION_ID = version;
+
+    String serviceId = null;
+
+    Path appYamlPath = MavenTestingUtils.getProjectFilePath("src/main/appengine/app.yaml");
```

Comment:
It was used to query the deployed application in remote testing scenarios. (both from an HTTP point of view, and also the various gcloud commands)